appservice/src/tree/AppSettingsTreeItem.ts

```diff
@@ -57,11 +57,18 @@ export class AppSettingsTreeItem extends AzureParentTreeItem<ISiteTreeRoot> {
 
     public async loadMoreChildrenImpl(_clearCache: boolean): Promise<AzureTreeItem<ISiteTreeRoot>[]> {
         this._settings = await this.root.client.listApplicationSettings();
+        const slotConfigurationSettings: SlotConfigNamesResource = await this.root.client.listSlotConfigurationNames();
         const treeItems: AppSettingTreeItem[] = [];
         // tslint:disable-next-line:strict-boolean-expressions
         const properties: { [name: string]: string } = this._settings.properties || {};
         Object.keys(properties).forEach((key: string) => {
-            treeItems.push(new AppSettingTreeItem(this, key, properties[key], this._commandId));
+            const appSettingTreeItem: AppSettingTreeItem = new AppSettingTreeItem(this, key, properties[key], this._commandId);
+            if (slotConfigurationSettings.appSettingNames && slotConfigurationSettings.appSettingNames.find((value: string) => {
```

Comment:
This code is a duplicate of `refreshImpl`. I prefer this pattern:
https://github.com/Microsoft/vscode-azurestorage/blob/master/src/azureStorageExplorer/storageAccounts/storageAccountNode.ts#L63

Basically you make a static `createAppSettingTreeItem` and make the constructor private to ensure that `refreshImpl` is always called when the tree item is created.

Reply:
Hmm, one issue with doing it like that is I'm going to have to call `const slotSettings: SlotConfigNamesResource = await this.root.client.listSlotConfigurationNames();` each time a child is created vs. the one time (and then once per refresh).  Should I store that on the parent and refresh it whenever it's updated?

Reply:
If you store it in the parent you just have to make sure you update it when the child is refreshed.

Or you could just make `slotSettings` an optional parameter to `refreshImpl`.

Or you could just leave it as is - idc